client/src/components/dom/Auth/AuthModal/AuthModal.tsx

```diff
@@ -0,0 +1,246 @@
+import { useState } from "react";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import styles from "./AuthModal.module.scss";
+import { AuthAction } from "../Auth";
+
+type FormDetails = {
+  email: string;
+  password: string;
+  confirm: string;
+};
+type FormErrors = {
+  email: string;
+  password: string;
+  confirm: string;
+};
+export default function AuthModal(props: {
+  authAction: AuthAction;
+  setAuthAction: (action: AuthAction) => void;
+}) {
+  /*COMPONENT STATE*/
+  const [formDetails, setFormDetails] = useState<FormDetails>({
+    email: "",
+    password: "",
+    confirm: "",
+  });
+
+  const [formErrors, setFormErrors] = useState<FormErrors>({
+    email: "",
+    password: "",
+    confirm: "",
+  });
+  const supabase = useSupabaseClient();
+
+  if (!supabase) return;
+
+  /*COMPONENT FUNCTIONS*/
+  function handleInputChange(e: any) {
+    setFormDetails({ ...formDetails, [e.target.name]: e.target.value });
+  }
+
+  function resetForm() {
+    setFormDetails({ email: "", password: "", confirm: "" });
+  }
+  function resetErrors() {
+    setFormErrors({ email: "", password: "", confirm: "" });
+  }
+
+  function switchActions() {
+    resetForm();
+    resetErrors();
+    props.setAuthAction(
+      props.authAction === AuthAction.Login ? AuthAction.Register : AuthAction.Login,
+    );
+  }
+
+  function closeModal(e: any) {
+    if (e.target === e.currentTarget) {
+      resetForm();
+      props.setAuthAction(AuthAction.Idle);
+    }
+  }
+
+  function verifyLoginInputs() {
+    let isValid = true;
+    resetErrors();
+
+    // Check if email is valid
+    if (!formDetails.email.includes("@")) {
+      isValid = false;
+      setFormErrors(prev => ({ ...prev, email: "Invalid email" }));
+    }
+    if (formDetails.password === "") {
+      isValid = false;
+      setFormErrors(prev => ({ ...prev, password: "Required" }));
+    }
+    return isValid;
+  }
+
+  function verifyRegisterInputs() {
+    let isValid = true;
+    resetErrors();
+
+    // Check password length
+    if (formDetails.password.length < 6) {
+      isValid = false;
+      setFormErrors(prev => ({ ...prev, password: "Password must be at least 6 characters" }));
+    }
+    // Check if passwords match
+    if (formDetails.password !== formDetails.confirm) {
+      isValid = false;
+      setFormErrors(prev => ({ ...prev, confirm: "Passwords do not match" }));
+    }
+    // Check if email is valid
+    if (!formDetails.email.includes("@")) {
+      isValid = false;
+      setFormErrors(prev => ({ ...prev, email: "Invalid email" }));
+    }
+    // Check if any fields are empty
+    for (const key in formDetails) {
```

Comment:
A bit of a nit-pick, but iterating through an object has more ergonomic options:

```ts
if (Object.values(formDetails).some(value => value === "")) {
	// handle error context
}
```

or

```ts
Object.values(formDetails).forEach( value => {
	if (value === "") {
		// handle error context
	}
});
```

if you really need the key:
```ts
Object.entries(formDetails).forEach((key, value) => {...});
```